File: backend/entity/equipment.go
package entity
import "gorm.io/gorm"

type Equipment struct{
	gorm.Model
	EquipmentName string `json:"equipment_name"`

	EquipmentTotalQuantity     uint    `json:"total_quantity"`
	EquipmentRemainingQuantity uint    `json:"remaining_quantity"`
	EquipmentUsedQuantity      uint    `json:"used_quantity"`
	
	EquipmentTypeID uint `json:"equipment_type_id"`
	EquipmentType *EquipmentType `gorm:"foreignKey: equipment_type_id" json:"equipment_type"`


	StageEquipments []StageEquipment `gorm:"foreignKey:EquipmentID" json:"stage_equipments"`

}

File: frontend/src/interface/equipment.ts
import type { StageInterface } from "./stage";


export interface EquipmentInterface {
  ID?: number; // สำหรับสร้างใหม่อาจยังไม่มี ID
  equipment_name: string;
  equipment_type: EquipmentTypeInterface;

  total_quantity: number;      // จำนวนทั้งหมด
  remaining_quantity: number;  // จำนวนเหลือใช้งาน
  used_quantity?: number;      // จำนวนที่ถูกใช้ (optional, backend คำนวณ)
  
  // สำหรับ Stage Assignment
  stage_equipments?: StageEquipmentInterface[];
}

export interface EquipmentTypeInterface {
  ID: number;
  equipment_type: string; // เช่น Sound, Light, Stage
}

export interface StageEquipmentInterface {
  stage_id: number;
  equipment_id: number;
  stage_quantity: number;
  equipment: EquipmentInterface;
  stage: StageInterface;
}




